### PR TITLE
fix global flag fail in Regex by resetting last Index

### LINKED ISSUE #37890: global flag in regular expression fails test
**Link to the challenge:** https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters

## Issue

Adding a global flag to the regular expression has the test fail for the first two assertions.

```js
let myString = "Eleanor Roosevelt";
let myRegex = /(Franklin|Eleanor).*Roosevelt/g;
let result = myRegex.test(myString);
```

This can be confusing considering also how the challenge is introduced, since the code snippet explaining the concept shows the global flag being used.

```js
let testStr = "Pumpkin";
let testRegex = /P(engu|umpk)in/g;
testRegex.test(testStr);
```

## Suggestion

Update the code snippet introducing the challenge:

```diff
-let testRegex = /P(engu|umpk)in/g;
+let testRegex = /P(engu|umpk)in/;
```

However, since the global flag should not affect the outcome of the test (to what I can ascertain), a better suggestion would be to update the test itself, to account for its presence.

### PR DESCRIPTION
**Pull request by yyazdi13**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #37890 

Hi I'm pretty new at this so I just tried to follow along with a similar solution used in PR #37941. Please let me know if I need to improve anything

### CODE REVIEW

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for making a PR for this. Could you also please add this to the final 2 tests, and, while we are at it, could you fix the final test so it matches the `text`?

Instead of `"Can me if you can."` -> `"Catch me if you can."`

@ojeytonwilliams In the issue, you mentioned adding this to the `js-teardown`, do you still think that is would be beneficial to do? Or, just move on with this implementation?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-anything-with-wildcard-period.english.md [hunk 1]
```diff
@@ -36,17 +36,17 @@ tests:
...
-    testString: assert(unRegex.test("Seven days without a pun makes one weak."));
+    testString: unRegex.lastIndex = 0; assert(unRegex.test("Seven days without a pun makes one weak."));
   - text: Your regex <code>unRegex</code> should match <code>"nun"</code> in <code>"One takes a vow to be a nun."</code>
-    testString: assert(unRegex.test("One takes a vow to be a nun."));
+    testString: unRegex.lastIndex = 0; assert(unRegex.test("One takes a vow to be a nun."));
   - text: Your regex <code>unRegex</code> should match <code>"bun"</code> in <code>"She got fired from the hot dog stand for putting her hair in a bun."</code>
-    testString: assert(unRegex.test("She got fired from the hot dog stand for putting her hair in a bun."));
+    testString: unRegex.lastIndex = 0; assert(unRegex.test("She got fired from the hot dog stand for putting her hair in a bun."));
   - text: Your regex <code>unRegex</code> should not match <code>"There is a bug in my code."</code>
     testString: assert(!unRegex.test("There is a bug in my code."));
```
[the remark this answers is not in the document]
Comment (yyazdi13) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-anything-with-wildcard-period.english.md]: okay no problem, I will take a look

**Review by ojeytonwilliams — APPROVED** [review]
@Sky020 it would be cleaner to do it in the teardown, but this is fine.

LGTM :+1:

**Review by ShaunSHamilton — APPROVED** [review]
I am happy with moving this forward 👍

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**